src/Pact/Types/Term.hs

```diff
@@ -1214,11 +1207,31 @@ pattern TLitInteger i <- TLiteral (LInteger i) _
 pattern TLitBool :: Bool -> Term t
 pattern TLitBool b <- TLiteral (LBool b) _
 
-
 tLit :: Literal -> Term n
 tLit = (`TLiteral` def)
 {-# INLINE tLit #-}
 
+$( return [] )
+instance Eq n => Ord (Def n) where
+  a `compare` b = nm a `compare` nm b
+    where nm d = (_dModule d, _dDefName d)
+
+deriving instance Show n => Show (Def n)
```

Comment:
I need to understand how you have determined that using `deriving` up above is a problem. Do you have code that actually breaks under some GHC version? Please provide compiler output.

Reply:
It breaks with GHC-8.10. The links in the the PR description and in the commit messages provide some background. The issue are the recursive dependency between the instances.

Reply:
If there isn't a reason why these are here, I would prefer to keep these with the types as is idiomatic.